receiver/splunksearchapireceiver/receiver.go

```diff
@@ -67,66 +67,83 @@ func (ssapir *splunksearchapireceiver) runQueries(ctx context.Context) error {
 			ssapir.logger.Error("error polling for search completion", zap.Error(err))
 		}
 
-		// fetch search results
-		ssapir.logger.Info("fetching search results")
-		results, err := ssapir.getSplunkSearchResults(searchID)
-		if err != nil {
-			ssapir.logger.Error("error fetching search results", zap.Error(err))
-		}
-		ssapir.logger.Info("search results fetched", zap.Int("num_results", len(results.Results)))
+		var resultCountTracker = 0 // track number of results exported
+		var offset = 0             // offset for pagination
+		var limitReached = false
+		for {
+			ssapir.logger.Info("fetching search results")
+			results, err := ssapir.getSplunkSearchResults(searchID, offset)
+			if err != nil {
+				ssapir.logger.Error("error fetching search results", zap.Error(err))
+			}
+			ssapir.logger.Info("search results fetched", zap.Int("num_results", len(results.Results)))
 
-		// parse time strings to time.Time
-		earliestTime, err := time.Parse(time.RFC3339, search.EarliestTime)
-		if err != nil {
-			// should be impossible to reach with config validation
-			ssapir.logger.Error("earliest_time failed to be parsed as RFC3339", zap.Error(err))
-		}
+			// parse time strings to time.Time
+			earliestTime, err := time.Parse(time.RFC3339, search.EarliestTime)
+			if err != nil {
+				ssapir.logger.Error("earliest_time failed to be parsed as RFC3339", zap.Error(err))
+			}
 
-		latestTime, err := time.Parse(time.RFC3339, search.LatestTime)
-		if err != nil {
-			// should be impossible to reach with config validation
-			ssapir.logger.Error("latest_time failed to be parsed as RFC3339", zap.Error(err))
-		}
+			latestTime, err := time.Parse(time.RFC3339, search.LatestTime)
+			if err != nil {
+				ssapir.logger.Error("latest_time failed to be parsed as RFC3339", zap.Error(err))
+			}
 
-		logs := plog.NewLogs()
-		for idx, splunkLog := range results.Results {
-			if idx >= search.Limit && search.Limit != 0 {
-				break
+			logs := plog.NewLogs()
+			for idx, splunkLog := range results.Results {
+				if (idx+resultCountTracker) >= search.Limit && search.Limit != 0 {
+					limitReached = true
+					break
+				}
+				// convert log timestamp to ISO8601 (UTC() makes RFC3339 into ISO8601)
+				logTimestamp, err := time.Parse(time.RFC3339, splunkLog.Time)
+				if err != nil {
+					ssapir.logger.Error("error parsing log timestamp", zap.Error(err))
+					break
+				}
+				if logTimestamp.UTC().After(latestTime.UTC()) {
+					ssapir.logger.Info("skipping log entry - timestamp after latestTime", zap.Time("time", logTimestamp.UTC()), zap.Time("latestTime", latestTime.UTC()))
+					// logger will only log up to 10 times for a given code block, known weird behavior
+					// TODO: Consider breaking, assuming all logs are in order
```

Comment:
Fair thing to look in to; we'd need to make sure that the events being retrieved from Splunk are in order, which I'm not confident to say one way or another on quite yet

Reply:
Took some time to look into this, the pattern I'm seeing is that events are returned latest to earliest, which makes sense based on how the Splunk search UI works. So it would be correct to break once a log is earlier than the earliest time.